autotest/pyscripts/test_gdal_utils.py

```diff
@@ -149,6 +151,27 @@ def test_utils_np_arrays():
             assert isinstance(arr, array_util.ArrayLike.__args__)
 
 
+def test_utils_color_files():
+    """ test color palettes: read QML and TXT files """
+    items = [
+        dict(name='color_paletted_red_green_0-255.qml', count=256, pal={0: 0x00ffffff, 1: 0xFF808080}),
+        dict(name='colro_pseudocolor_spectral_0-100.qml', count=5, pal={0: 0xFFD7191C, 25: 0xFFFFFFBF}),
+    ]
+    root = Path(test_py_scripts.get_data_path('utilities'))
+    for item in items:
+        path = root / item['name']
+        path2 = path.with_suffix('.txt')
+        cp1 = ColorPalette()
+        cp2 = ColorPalette()
+        cp1.read_file(path)
+        # cp1.write_file(path2)
+        cp2.read_file(path2)
+        assert cp1 == cp2
+        assert len(cp1.pal) == item['count']
+        for k, v in item['pal'].items():
```

Comment:
couldn't that be just written as ``assert cp1.pal = item['pal']`` ?

Reply:
I didn't hardcode the entire palette in item['pal'], I thought the first two values should be enough.
I'll add a comment to explain that.